### DISCUSSION TITLE
Dev Question: How do I test my local changes...?

### DISCUSSION THREAD

**Discussion by mihawk90** [pyload/pyload#4391] [Help] [answered]
I'm a little stumped... I know my way around Python a little bit but I can't figure out how to run pyload from the git repository after cloning... I can't seem to find any executable or rather main run script.
The [CONTRIBUTING.md](https://github.com/pyload/pyload/blob/main/CONTRIBUTING.md#make-changes) is unfortunately really not helpful at all. It points out how to clone (I know), and to "start with your changes", then goes on to committing and PRing these changes. But no word about testing them anywhere.

I had to dig through the GitHub Actions that apparently this needs to be built/compiled (which is news to me, I have always ever only handled `.py` files):
https://github.com/pyload/pyload/blob/a65a968696f85c52c1f835d6d5dc22fd52fca12a/.github/workflows/build.yml#L53-L57

So I did that locally but I'm no step further. The file in `dist` is a binary that can't be executed:
```
z% chmod +x pyload_ng-0.5.0b3-py3-none-any.whl 
z% ./pyload_ng-0.5.0b3-py3-none-any.whl 
zsh: exec format error: ./pyload_ng-0.5.0b3-py3-none-any.whl
```
And nothing in `build` sticks out as runnable either:
```
z% pwd
/home/tarulia/Development/pyload/build/lib/pyload
z% python __main__.py
Traceback (most recent call last):
  File "/home/tarulia/Development/pyload/build/lib/pyload/__main__.py", line 11, in <module>
    from . import __version__
ImportError: attempted relative import with no known parent package
z% python __init__.py
/home/tarulia/Development/pyload/build/lib/pyload/__init__.py:15: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
  ```

When I install pyload-ng via pip I get an executable:
```
z% cat $(which pyload)
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import sys
from pyload.__main__ import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```
But no such file can be found in either `build` or `dist`, and the only place where contents of that file show up is `setup.py`:
```
z% git grep pyload.__main__
setup.cfg:    pyload = pyload.__main__:main
```
... which we used to compile with no results before.

So... I'm sorry but how do I go about this?
It seems like contributing is not made easy on this...

**Comment by GammaC0de** [comment, marked as answer]
first do `pip install --pre pyload-ng[all]` to install pyload and it depndencies.

to run pyload use the command `python3 -m pyload` from within the `src` directory after cloning the git repo

**Reply by mihawk90** [reply]
Thank you, lot easier then I thought it would be :)
